scipy/stats/_variation.py

```diff
@@ -91,31 +94,38 @@ def variation(a, axis=0, nan_policy='propagate', ddof=0, *, keepdims=False):
     array([1.05109361, 0.31428986, 0.146483  ])
 
     """
+    xp = array_namespace(a)
+    a = xp.asarray(a)
     # `nan_policy` and `keepdims` are handled by `_axis_nan_policy`
+    # `axis=None` is only handled for NumPy backend
+    if axis is None:
+        a = xp.reshape(a, (-1,))
+        axis = 0
+
     n = a.shape[axis]
     NaN = _get_nan(a)
 
     if a.size == 0 or ddof > n:
         # Handle as a special case to avoid spurious warnings.
         # The return values, if any, are all nan.
-        shp = np.asarray(a.shape)
-        shp = np.delete(shp, axis)
-        result = np.full(shp, fill_value=NaN)
-        return result[()]
+        shp = list(a.shape)
+        shp.pop(axis)
+        result = xp.full(shp, fill_value=NaN)
+        return result[()] if result.ndim == 0 else result
 
-    mean_a = a.mean(axis)
+    mean_a = xp.mean(a, axis=axis)
 
     if ddof == n:
```

Comment:
@lucascolley Could you take a look at this for `ddof == n`? I can imagine that we could go this in a JAX-compatible way (e.g. using `where` instead of assignment). _Should_ we do that, or someday will we be able to use the existing style again (or something like it)?

Reply:
That depends on https://github.com/data-apis/array-api/issues/609, which hasn't been particularly active recently. I'm not sure how inconvenient it should be to merit skipping rather than fixing - Ralf made the calls over on my PR.

Reply:
I made the change to `xp.where` anyway since it was straightforward